### DISCUSSION TITLE
Why SelectedItem of ComboBox is null in DataGridColumn?

### DISCUSSION THREAD

**Discussion by alvahtin** [AvaloniaUI/Avalonia#15192] [Q&A] [answered]
I use DataGrid

```
         <DataGrid ItemsSource="{Binding SinParamsList}" SelectionMode="Single" Name="dgSinParams" GridLinesVisibility="All">
		<DataGrid.Columns>
			<DataGridTextColumn Header="Amplitude" Binding="{Binding Amplitude}" Width="*" IsReadOnly="False" />
			<DataGridTemplateColumn Header="Frequency" Width="*">
				<DataGridTemplateColumn.CellTemplate>
					<DataTemplate DataType="model:SinParams">
						<TextBlock Text="{Binding Frequency}" />
					</DataTemplate>
				</DataGridTemplateColumn.CellTemplate>
				<DataGridTemplateColumn.CellEditingTemplate>
					<DataTemplate DataType="model:SinParams">
						<ComboBox SelectedItem="{Binding Frequency}" ItemsSource="{Binding $parent[UserControl].FrequencyList}" HorizontalAlignment="Stretch" Background="White" GotFocus="ComboBox_GotFocus" />
					</DataTemplate>
				</DataGridTemplateColumn.CellEditingTemplate>
			</DataGridTemplateColumn>
		</DataGrid.Columns>
	</DataGrid>
```

When I select a cell with the ComboBox to edit, SelectedItem is null. But value, wich binding to SelectedItem is in the ItemsSource list. And when I select value from Items, it is set to binding SinParams value. And when I select a value from Items, it is set to the bound SinParams value. But when I lost foucus and select the cell again, the SelectedItem is null. But when I lost focus and select this cell, SelectedItem is null again.

https://github.com/AvaloniaUI/Avalonia/assets/50268307/ef5dab34-44b7-45b0-89bb-eea3346e9a51

I tried setting Mode=TwoWay for SelectedItem binding but it doesn't help. What am I doing wrong?

**Comment by timunie** [comment, marked as answer]
1. Set `ItemsSource` before settings `SelectedItem` in order to make sure items are there
2. Give your UserControl a name `x:Name="root"` and bind like that: `ItemsSource="{Binding #root.FrequencyList}"`
3. Are you sure `FrequencyList` is part of the UserControl rather than it's ViewModel?

**Comment by alvahtin** [comment]
Amazing! The p. 2 worked. Should use `ItemsSource="{Binding #root.FrequencyList}"` instead `ItemsSource="{Binding $parent[UserControl].FrequencyList}"`

Thank you, very much! I don’t know why, but I will know this feature.

**Reply by timunie** [reply]
$parent[] uses the logical tree to find the anchor. But in DataGrid, a cell isn't part of the logical tree so it cannot find the logical parent.

**Reply by alvahtin** [reply]
But `FrequencyList` is getting bound. It is detected in UserControl. Probably, binding `ItemsSource="{Binding $parent[UserControl].FrequencyList}"` is executed later than `SelectedItem="{Binding Frequency}"`. But `ItemsSource="{Binding #root.FrequencyList}"` at the same time.

**Reply by timunie** [reply]
No I guess what happens here, it is a property of the ViewModel, aka DataContext and not the control itself. So bascially: 

`{Binding #rrot.((vm:MainViewModel)DataContext).MyList}`

**Reply by alvahtin** [reply]
No. MainViewModel doesn't have FrequencyList. It is in UserControl.
